File: src/server/bo/GruppenTeilnahme.py
from server.bo import BusinessObject as bo


class GruppenTeilnahme(bo.BusinessObject):
    """Realisierung der Klasse GruppenTeilnahme zur Zuordnung von Personen zu Gruppen"""

    def __init__(self):
        super().__init__()
        self._person_id = 0  # Fremdschlüsselbeziehung zur Person, welche an einer Lerngruppe teilnimmt
        self._gruppen_id = (
            0  # Fremdschlüsselbeziehung zur Lerngruppe, an welcher die Person teilnimmt
        )
        self._ist_admin = False

    def get_person_id(self):
        """Auslesen der Person ID"""
        return self._person_id

    def set_person_id(self, person_id: int):
        """Setzen der Person ID"""
        self._person_id = person_id

    def get_gruppen_id(self):
        """Auslesen der Gruppen ID"""
        return self._gruppen_id

    def set_gruppen_id(self, gruppen_id: int):
        """Setzen der Gruppen ID"""
        self._gruppen_id = gruppen_id

    def get_ist_admin(self):
        """Auslesen des boolschen Wertes, ob eine Person der Admin einer Gruppe ist"""
        return self._ist_admin

    def set_ist_admin(self, wert: bool):
        """Setzen des boolschen Wertes, ob eine Person der Admin einer Gruppe ist"""
        self._ist_admin = wert

    @staticmethod
    def from_dict(dictionary=dict()):
        obj = GruppenTeilnahme()
        obj.set_id(dictionary["id"])
        obj.set_erstellungszeitpunkt(dictionary["erstellungszeitpunkt"])
        obj.set_person_id(dictionary["person_id"])
        obj.set_gruppen_id(dictionary["gruppen_id"])
        obj.set_ist_admin(dictionary["ist_admin"])
        return obj

File: frontend/src/api/GruppenTeilnahme.js
import BusinessObject from "./BusinessObject";


export default class GruppenTeilnahme extends BusinessObject {
    /**
       * Erstellt ein BO einer GruppenTeilnahme mit den gegebenen Parametern.
       *
       * @param person_id - Die ID der Person.
       * @param gruppen_id - Die ID der Gruppe.
       * @param ist_admin - Überprüfung des Administrator Status der Person.
     */

    constructor(person_id, gruppen_id, ist_admin) {
        super();
        this.person_id = person_id;
        this.group_id = gruppen_id;
        this.ist_admin = ist_admin;
    }

    /**
     * Liest die person_id aus.
     */
    get_person_id() {
        return this.person_id;
    }

    /**
     * Legt die person_id fest.
     */
    set_person_id(person_id=0) {
        this.person_id = person_id;
    }

    /**
     * Liest die gruppen_id aus.
     */
    get_gruppen_id() {
        return this.group_id;
    }

    /**
     * Legt die gruppen_id fest.
     */
    set_gruppen_id(gruppen_id=0) {
        this.group_id = gruppen_id;
    }

    /**
     * Gibt den Administratorenstatus aus.
     */
    get_ist_admin() {
        return this.ist_admin;
    }

    /**
     * Legt den Administratorenstatus fest.
     */
    set_ist_admin(wert=Boolean) {
        this.ist_admin = wert;
    }

    /**
     * Gibt das BO als String zurück (Für Debugging Zwecke).
     */
    static fromJSON(gruppenteilnahme) {
        let result = [];

        if (Array.isArray(gruppenteilnahme)) {
            gruppenteilnahme.forEach((g) => {
                Object.setPrototypeOf(g, GruppenTeilnahme.prototype);
                result.push(g);
            })
        }   else {
            let g = gruppenteilnahme
            Object.setPrototypeOf(g, GruppenTeilnahme.prototype);
            result.push(g);
        }

        return result;
    }
}